syncd/Syncd.cpp

```diff
@@ -4309,6 +4309,13 @@ void Syncd::onSwitchCreateInInitViewMode(
                 sai_serialize_object_id(switchVid).c_str(),
                 newHw.c_str());
 
+        /*
+         * The line below is added due to a behavior change of SAI call.
+         *
+         * TODO: remove the line when SAI vendor agrees fix on their end.
+        */
+        currentHw = currentHw == "none"? "":currentHw;
```

Comment:
The coding style in this repo seems to require white space before and after ":".

Reply:
How come suddenly its none? Is it Sai internal vendor?

Reply:
You miss space on last comment linę before star

Reply:
> How come suddenly its none? Is it Sai internal vendor?

With new SAI code in 202205+ images the return value for GET call on hwinfo is `none`. We have confirmed that older SAI was returning `''`. This will be further  discussed with BRCM SAI team. But, in the meanwhile, we need this workaround to unblock/uncover other issues.

Reply:
@yxieca @kcudnik Addressed.